src/book/02-system/01-foundations/02-build-system.mdx

```diff
@@ -566,3 +566,61 @@ target = {
     "asm_object": "elf64",
 }
 ```
+
+### Dependency Declaration
+
+CMake needs to know what dependencies we have and where to find them.
+In a module's folder the `CMakeLists.txt` file tells CMake what is needed to build a module.
+The `nuclear_modlue()` function tells CMake that this directory contains a module that may be added to a role.
+This function may take named parameters.
+
+| Parameter  | Usage                                                                          |
+| ---------- | ------------------------------------------------------------------------------ |
+| LANGUAGE   | The Programming Language that this module is in (currently only c++)           |
+| LIBRARIES  | The libraries that this module depends on and must be linked with              |
+| SOURCES    | Extra code outside the `src` directory that must be built                      |
+| DATA_FILES | Extra data files outside the `data` directory that is necessary for the module |
```

Comment:
```suggestion
| DATA_FILES | Extra data files outside the `data` directory that the module depends on |
```
Not sure if this is right but the sentence felt incomplete/grammar dodgy

Reply:
```suggestion
| DATA_FILES | Extra data files outside the `data` directory that the module uses |
```